core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -78,6 +81,10 @@ public void accept(String file) {
   RemoveSnapshots(TableOperations ops) {
     this.ops = ops;
     this.base = ops.current();
+
+    ValidationException.check(
+        !PropertyUtil.propertyAsBoolean(base.properties(), SNAPSHOT, SNAPSHOT_DEFAULT),
+        "Not allowed to expire snapshots in a snapshot table as this may remove files in the original table");
```

Comment:
We really need better naming. This is extremely confusing :)

Reply:
"The target table was created using the snapshot action against another table which means it is not the exclusive owner of its data files. This means that we cannot determine if data files are actually no longer needed after running an expireSnapshots command and could unintentionally remove files needed by the original table." -- Maybe?

Reply:
I agree, let me update this.

Reply:
What about thinking about this as "un-gc'able" instead of specifically a snapshot table to be more generic?

Reply:
I think it is a good idea, @jacques-n. We will need a good name, though. Is `gc.enabled` descriptive enough? Any other alternatives?

Also, we should maybe respect this property while removing orphan files.

Reply:
I like the more specific gc enabled flag. That seems like a better way to do this.